src/main/java/net/snowflake/ingest/streaming/internal/FlushService.java

```diff
@@ -436,8 +437,10 @@ BlobMetadata buildAndUpload(String filePath, List<List<ChannelData<T>>> blobData
         // TODO: encryption is not yet supported by server side for Parquet
         long iv = curDataSize / Constants.ENCRYPTION_ALGORITHM_BLOCK_SIZE_BYTES;
         byte[] encryptedCompressedChunkData =
-            Cryptor.encrypt(
-                compressedAndPaddedChunkData, firstChannel.getEncryptionKey(), filePath, iv);
+            bdecVersion == Constants.BdecVersion.THREE
+                ? compressedAndPaddedChunkData
```

Comment:
I don't think we can send unencrypted data, have you talk to the our security partners about this?

Reply:
it is on the road map

Reply:
Good catch Toby - we need to ensure that this unencrypted data cannot be queried in Snowflake as supplied from the Client SDK. Let's either follow up with a PR to support encryption (with a JIRA linked here we can track) or add encryption support now. If the latter would dramatically increase the size of the PR I'd be happy filing a JIRA, adding it here, and following up immediately afterward.

Reply:
The encryption is not supported by the server side scanner, there is an internal JIRA subtask for this.

Reply:
update: @sfc-gh-azagrebin found out that we can use the encryption w/o file caching in XP (see changes in his last commit).